atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/utils/LivenessChecker.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.jepsen.utils;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.CheckerResult;
+import com.palantir.atlasdb.jepsen.ImmutableCheckerResult;
+import com.palantir.atlasdb.jepsen.events.Checker;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.events.EventVisitor;
+import com.palantir.atlasdb.jepsen.events.FailEvent;
+import com.palantir.atlasdb.jepsen.events.ImmutableInfoEvent;
+import com.palantir.atlasdb.jepsen.events.InfoEvent;
+import com.palantir.atlasdb.jepsen.events.InvokeEvent;
+import com.palantir.atlasdb.jepsen.events.OkEvent;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class LivenessChecker implements Checker {
+    private final Predicate<OkEvent> livenessJudgmentHandler;
+
+    public LivenessChecker(Predicate<OkEvent> livenessJudgmentHandler) {
+        this.livenessJudgmentHandler = livenessJudgmentHandler;
+    }
+
+    @Override
+    public CheckerResult check(List<Event> events) {
+        Visitor visitor = new Visitor(livenessJudgmentHandler);
+        events.forEach(event -> event.accept(visitor));
```

Comment:
If you're already doing this, why don't we make the visitor generic ("corrected visitor pattern") so you can return true when you want to stop and not have to go through the entire history when you probably know very quickly liveness is satisfied. Could also choose not to process entire history when we find a correctness case, but probably don't want to do that

Reply:
This is much better!